### PR TITLE
Update 6.2 tldr and add common example

### PR DESCRIPTION
**Pull request by animir**
[rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible) package works not only with Express.js and can be used in many cases to protect applications from DDoS and brute force attacks

### CODE REVIEW

**Comment by js-kyle** [comment]
Awesome! I will review this shortly

**Comment by i0natan** [comment]
@animir Welcome aboard. Seems like a great work over there.

- Does this package stand to our vendors inclusion principles here or provide some outstanding added-value over the recommended package?
- Why in the TL&DR is mentioned before the 'rate limiting middlewares exmaples', isn't it a middleware?

**Comment by animir** [comment]
@i0natan Thanks.

rate-limiter-flexible is not middleware or plugin. It provides classes and wrappers, which help to achieve any logic related to rate limiting.

1. rate-limiter-flexible is suitable for any framework or pure node.js application (not only express). 
2. it provides some useful options like insurance strategy, block strategy, leaky bucket analogy and other
3. TypeScript declaration bundled
4. What I really like about rate-limiter-flexible is ability to work on one server without any software in Cluster. It also supports PM2 launched applications. IPC is quite fast.
5. I thoroughly tested it and high traffic to handle all possible race conditions on different databases.
6. There is no prod dependencies. It is big plus considering what happened in 2018.
7. It is flexible. No preset headers. No callbacks. Ready for any idea. Developer can even use it with jobs queue with idea like 'generate 10 reports per hour or schedule for night time'.

**Comment by i0natan** [comment]
@animir Your package looks promising, great docs, great packaging and the care for small details can be easily inferred. I personally liked the 'In-memory Block Strategy'. 

All that said, we've a [vendors policy](https://github.com/i0natan/nodebestpractices/blob/master/.operations/writing-guidelines.md) that states that 750 downloads/daily and being within the top 3 in search results is mandatory for inclusion in this list. It aims to ensure that all the recommendations here are for highly reputable and battle-tested packages. Unfortunately, your package still doesn't meet this guideline.

Based on the quality of work I'm highly positive that it will reach these goals, we can contribute our modest help by Tweeting about it (I already star it...) so let's park this PR until it's eligible for merge?

p.s. Are you sure you protect against DDOS attacks and not DOS?

**Comment by animir** [comment]
@i0natan Strict rules. Ok, I do not want to break these rules, changes can wait. Just some thoughts (may be they will be interesting for community) :
1. Old packages have priorities even if they worse than new. it is actually a big problem of open source. Many developers just get the first package from search and additionally star it. I do the same, tbh. So "the top 3" rule limits progress overall. I would suggest to look into statistic in dynamic. For example, https://bestofjs.org/tags/middleware/trending/last-3-months shows trending "middlewares" for the last 3 months. And it is clear that "rate-limiter-flexible" is getting about 0.8 stars every day, which is good result considering that it is new and quite specific package. 

2. expressjs.com already added "rate-limiter-flexible" to [best security practice section](https://expressjs.com/en/advanced/best-practice-security.html#additional-considerations). I think. it is a good sign too. Subjective opinion of known and experienced developer or group should be stronger than simple star on github. This index of confidence may be considered too.

At the end of day, those 2 points would be better to implement on github: show stars number in dynamic and consider who gives stars. It is strange, if Yoni Goldberg's star is the same strong as first year student's star.

> p.s. Are you sure you protect against DDOS attacks and not DOS?

It can protect against both on highest OSI layers.

**Comment by i0natan** [comment]
(A message to bot below)

/remind me in to comment in 3 days

**Comment by i0natan** [comment]
@animir I absolutely understand your concerns, the "rating" world is not accurate or fair, but we need some protection from inferior/unmaintained/immature packages (not yours of course) and I can't think of a better one than real-world usage and downloads. We would love to collaborate with you on security issues, help improving/promoting this package and do things together until this package reach the desired numbers... :]

**Comment by i0natan** [comment]
@animir Let's revisit this PR in 3 months?

**Comment by animir** [comment]
@i0natan I noticed, that there was merge conflict and fixed it, so changes are more in sync with master.

I'll post here the next time, when package fits requirements, ok?
It's still downloaded about 600 per day :-)

**Comment by animir** [comment]
@i0natan `rate-limiter-flexible` is downloaded ~800 times per day for the last 2 weeks.
It is on the first place in the search by relevant phrase `node ddos protection`.

I've simplified an example slightly, please, review changes.

File: sections/security/limitrequests.md [hunk 1]
```diff
@@ -2,7 +2,45 @@
 
 ### One Paragraph Explainer
 
-Rate limiting should be implemented in your application to protect a Node.js application from being overwhelmed by too many requests at the same time. Rate limiting is a task best performed with a service designed for this task, such as nginx, however it is also possible with application middleware such as [express-rate-limiter](https://www.npmjs.com/package/express-rate-limit).
+Rate limiting should be implemented in your application to protect a Node.js application from being overwhelmed by too many requests at the same time. Rate limiting is a task best performed with a service designed for this task, such as nginx, however it is also possible with [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible) package or  middleware such as [express-rate-limiter](https://www.npmjs.com/package/express-rate-limit) for Express.js applications.
```
Comment (js-kyle) [inline sections/security/limitrequests.md]:
```suggestion
Rate limiting should be implemented in your application to protect a Node.js application from being overwhelmed by too many requests at the same time. Rate limiting is a task best performed with a service designed for this task, such as nginx, however it is also possible with [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible) package or middleware such as [express-rate-limiter](https://www.npmjs.com/package/express-rate-limit) for Express.js applications.
```

File: sections/security/limitrequests.md [hunk 2]
```diff
@@ -2,7 +2,45 @@
 
 ### One Paragraph Explainer
 
-Rate limiting should be implemented in your application to protect a Node.js application from being overwhelmed by too many requests at the same time. Rate limiting is a task best performed with a service designed for this task, such as nginx, however it is also possible with application middleware such as [express-rate-limiter](https://www.npmjs.com/package/express-rate-limit).
+Rate limiting should be implemented in your application to protect a Node.js application from being overwhelmed by too many requests at the same time. Rate limiting is a task best performed with a service designed for this task, such as nginx, however it is also possible with [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible) package or  middleware such as [express-rate-limiter](https://www.npmjs.com/package/express-rate-limit) for Express.js applications.
+ 
+  ### Code example: pure NodeJS app with [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible)
```
Comment (js-kyle) [inline sections/security/limitrequests.md]:
```suggestion
  ### Code example: pure Node.js app with [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible)
```

File: sections/security/limitrequests.md [hunk 3]
```diff
@@ -2,7 +2,45 @@
...
+   blockDuration: 2, // block for 2 seconds if consumed more than 20 points per second
+ });
+ 
+ http.createServer((req, res) => {
+   rateLimiter.consume(req.socket.remoteAddress)
+     .then((rateLimiterRes) => {
+        // Some app logic here
+ 
+        res.writeHead(200);
+        res.end()
```
Comment (js-kyle) [inline sections/security/limitrequests.md]:
```suggestion
        res.end();
```

File: sections/security/limitrequests.md [hunk 4]
```diff
@@ -2,7 +2,45 @@
...
+      })
+      .catch(() => {
+        res.writeHead(429);
+        res.end('Too Many Requests')
```
Comment (js-kyle) [inline sections/security/limitrequests.md]:
```suggestion
        res.end('Too Many Requests');
```

**Comment by animir** [comment]
@js-kyle Thank you for comments, all requested changes are approved.

**Pull request merged**